eureka_ml_insights/metrics/nphard_sat_metrics.py

```diff
@@ -0,0 +1,44 @@
+from .metrics_base import Metric
+
+
+class NPHardSATMetric(Metric):
+    """
+    A metric class for evaluating solutions to the SAT.
+    A prediction is considered correct if it is a valid SAT assignment and matches one of the optimal solutions.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def is_sat_soln_present(self, optimal_assignment_curr, assignment_string):
```

Comment:
A doc string would be helpful to know what the optimal_assignment_list should contain. Does it include all permutations of variables? Should we worry about assignment orders not matching but being equivalent?